pkg/manager/router/backend_fetcher.go

```diff
@@ -121,8 +121,11 @@ func (pf *PDFetcher) fetchBackendList(ctx context.Context) {
 		if response, err = pf.client.Get(ctx, infosync.TopologyInformationPath, clientv3.WithPrefix()); err == nil {
 			break
 		}
-		pf.logger.Error("fetch backend list failed, will retry later", zap.Error(err))
-		time.Sleep(pf.config.RetryInterval)
+		// Ignore errors when TiProxy shuts down.
+		if ctx.Err() == nil {
```

Comment:
Maybe just move `if ctx.Err != nil { return }` here?

Reply:
I changed it to `if ctx.Err != nil { return } `. But I can't remove the following `if ctx.Err != nil { return } `.